MetaMask/metamask-extension
# Fix custom nonce support + incoming transaction display

## **Description**

### Nonce support

Fix the custom nonce support after migrating to the core `TransactionController`.

This was caused due to the extension using an additional metadata property `customNonceValue` that wasn't supported in the core controller which expected custom nonces to be set in the `txParams.nonce` property.

### Incoming transactions

Fix the display of incoming transactions in the activity list.

This was caused due to the missing `type` property on incoming transaction entries in the state.

### Notes

This is solely a dependency update with no code changes.

See [here](https://github.com/MetaMask/core/pull/3579) for relevant updates in the `@metamask/transaction-controller` package.

## **Related issues**

Fixes: #22043 #22045 

## **Manual testing steps**

1. Enable custom nonces in settings.
2. Verify error with low custom nonce.
3. Verify queued transaction with high custom nonce.
4. Verify transactions with custom nonces disabled.
5. Verify all nonce related functionality such as speed up and cancel.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
